flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGate.java

```diff
@@ -524,6 +533,15 @@ private BufferOrEvent transformToBufferOrEvent(
 			boolean moreAvailable,
 			InputChannel currentChannel) throws IOException, InterruptedException {
 		if (buffer.isBuffer()) {
+			if (buffer.isCompressed()) {
```

Comment:
As I previously concerned, I tried to make the compression property as final in the constructor of `Buffer` instance. It makes sense to do that on producer side, but seems a bit trouble for consumer side because some original requested buffers are from `LocalBufferPool`.

So I have another option to avoid injecting the compression property in `BufferResponse` and `NetworkBuffer` on consumer side, as long as we judge this condition via `bufferDecompressor != null` instead here. I guess it makes sense to do that because whether to generate `bufferDecompressor` in `SingleInputGateFactory` also from the global configuration for enabling compression or not.